swap_meet/vendor.py

```diff
@@ -1,2 +1,99 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except: 
+            return False
+
+    def get_by_category(self, type):
+        '''
+        returns a list of Items in the inventory with the same category as 'type'
+        '''
+        item_list = []
+        for item in self.inventory:
+            if type == item.category:
+                item_list.append(item)
+        return item_list
+
+    def swap_items(self, vendor, my_item, their_item):
+
+        if my_item in self.inventory and their_item in vendor.inventory:
+            vendor.inventory.append(my_item)
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            vendor.inventory.remove(their_item)
+            return True
+        else: 
+            return False
+
+    def swap_first_item(self, vendor):
```

Comment:
You already have a function that will do this work for you!` swap_items() `can be called here and then you would pass in the first item of the vender and the other vendor as arguments
`return self.swap_items(other, self.inventory[0], other.inventory[0])`